### ISSUE TITLE
🔥 [ANR] Broadcast of Intent { act=com.google.android.c2dm.intent.RECEIVE pkg=com.test.app cmp=com.test.app/com.google.firebase.iid.FirebaseInstanceIdReceiver (has extras) }

### ISSUE THREAD

**Issue by RodolfoGS** [invertase/react-native-firebase#3900]
## Issue

I have a lot of ANRs in the Google Play Console.
```
Broadcast of Intent { act=com.google.android.c2dm.intent.RECEIVE flg=0x11000010 pkg=com.test.app cmp=com.test.app/com.google.firebase.iid.FirebaseInstanceIdReceiver (has extras) }

com.google.firebase.iid.FirebaseInstanceIdReceiver
```

I don't know how to reproduce, it just appear in my Google Play Console.

```
"main" prio=5 tid=1 TimedWaiting
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x73e7a890 self=0x7a43814c00
  | sysTid=23454 nice=0 cgrp=default sched=0/0 handle=0x7ac92d1548
  | state=S schedstat=( 1008328963 377962851 1610 ) utm=16 stm=84 core=7 HZ=100
  | stack=0x7ff4dbd000-0x7ff4dbf000 stackSize=8MB
  | held mutexes=
  at java.lang.Object.wait (Object.java)
- waiting on <0x08ca1456> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
- locked <0x08ca1456> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.parkNanos (LockSupport.java:230)
  at java.util.concurrent.FutureTask.awaitDone (FutureTask.java:447)
  at java.util.concurrent.FutureTask.get (FutureTask.java:205)
  at com.crashlytics.android.core.CrashlyticsCore.finishInitSynchronously (CrashlyticsCore.java)
  at com.crashlytics.android.core.CrashlyticsCore.access$100 (CrashlyticsCore.java)
  at com.crashlytics.android.core.CrashlyticsCore.doLog (CrashlyticsCore.java)
  at com.crashlytics.android.core.CrashlyticsCore.isBuildIdValid (CrashlyticsCore.java)
  at com.crashlytics.android.core.CrashlyticsCore.log (CrashlyticsCore.java)
  at com.crashlytics.android.core.CrashlyticsCore.logException (CrashlyticsCore.java)
  at com.crashlytics.android.core.CrashlyticsCore.onPreExecute (CrashlyticsCore.java)
  at com.crashlytics.android.core.CrashlyticsCore.setCrashlyticsNdkDataProvider (CrashlyticsCore.java)
  at com.crashlytics.android.core.CrashlyticsCore.setString (CrashlyticsCore.java)
  at com.crashlytics.android.core.CrashlyticsCore.onPreExecute (CrashlyticsCore.java)
  at io.fabric.sdk.android.InitializationTask.onPreExecute (InitializationTask.java)
  at io.fabric.sdk.android.services.concurrency.AsyncTask.access$200 (AsyncTask.java)
  at io.fabric.sdk.android.services.concurrency.AsyncTask.access$300 (AsyncTask.java)
  at io.fabric.sdk.android.services.concurrency.AsyncTask.doInBackground (AsyncTask.java)
  at io.fabric.sdk.android.services.concurrency.AsyncTask.executeOnExecutor (AsyncTask.java)
  at io.fabric.sdk.android.services.concurrency.PriorityAsyncTask.addDependency (PriorityAsyncTask.java)
  at io.fabric.sdk.android.services.concurrency.PriorityAsyncTask.addDependency (PriorityAsyncTask.java)
  at io.fabric.sdk.android.services.concurrency.PriorityAsyncTask.executeOnExecutor (PriorityAsyncTask.java)
  at io.fabric.sdk.android.services.concurrency.PriorityAsyncTask.isFinished (PriorityAsyncTask.java)
  at io.fabric.sdk.android.services.concurrency.PriorityAsyncTask.setError (PriorityAsyncTask.java)
  at io.fabric.sdk.android.services.concurrency.PriorityAsyncTask.setFinished (PriorityAsyncTask.java)
  at io.fabric.sdk.android.Kit.initialize (Kit.java)
  at io.fabric.sdk.android.Fabric.access$300 (Fabric.java)
  at io.fabric.sdk.android.Fabric.getExecutorService (Fabric.java)
  at io.fabric.sdk.android.Fabric.getKitMap (Fabric.java)
  at io.fabric.sdk.android.Fabric.initializeKits (Fabric.java)
  at io.fabric.sdk.android.Fabric.init (Fabric.java)
  at io.fabric.sdk.android.Fabric.access$100 (Fabric.java)
  at io.fabric.sdk.android.Fabric.getIdentifier (Fabric.java)
  at io.fabric.sdk.android.Fabric.setFabric (Fabric.java)
  at io.fabric.sdk.android.Fabric.extractActivity (Fabric.java)
  at io.fabric.sdk.android.Fabric.getKits (Fabric.java)
  at io.fabric.sdk.android.Fabric.with (Fabric.java)
- locked <0x00c10cd7> (a java.lang.Class<h.a.a.a.c>)
  at io.invertase.firebase.crashlytics.ReactNativeFirebaseCrashlyticsInitProvider.onCreate (ReactNativeFirebaseCrashlyticsInitProvider.java)
  at android.content.ContentProvider.attachInfo (ContentProvider.java:1936)
  at android.content.ContentProvider.attachInfo (ContentProvider.java:1911)
  at io.invertase.firebase.common.ReactNativeFirebaseInitProvider.attachInfo (ReactNativeFirebaseInitProvider.java)
  at android.app.ActivityThread.installProvider (ActivityThread.java:6447)
  at android.app.ActivityThread.installContentProviders (ActivityThread.java:5994)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:5900)
  at android.app.ActivityThread.access$1100 (ActivityThread.java:202)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1665)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:198)
  at android.app.ActivityThread.main (ActivityThread.java:6729)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:876)
```

Devices Affected:
![image](https://user-images.githubusercontent.com/870365/86807513-8f6d5080-c03f-11ea-9162-192acdd1976d.png)

---

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [x] my application is an AndroidX application?
- [x] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [x] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
System:
    OS: macOS 10.15.5
    CPU: (4) x64 Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz
    Memory: 48.75 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.4.0 - /usr/local/bin/node
    Yarn: 1.19.1 - /usr/local/bin/yarn
    npm: 6.14.5 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.5, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK:
      Android NDK: 21.3.6528147
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6514223
    Xcode: 11.5/11E608c - /usr/bin/xcodebuild
  npmPackages:
    react: ~16.9.0 => 16.9.0
    react-native: ~0.61.5 => 0.61.5
  npmGlobalPackages:
    react-native-cli: 2.0.1
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [x] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `@react-native-firebase/app: 7.3.1`

- **`Firebase` module(s) you're using that has the issue:**
  - `@react-native-firebase/admob: 7.2.2`
  - `@react-native-firebase/analytics: 7.1.6`
  - `@react-native-firebase/app: 7.3.1`
  - `@react-native-firebase/crashlytics: 7.1.7`
  - `@react-native-firebase/dynamic-links: 7.2.1`
  - `@react-native-firebase/iid: 7.1.7`
  - `@react-native-firebase/remote-config: 7.1.6`

- **Are you using `TypeScript`?**
  - `N`
- **Are you using `Expo`?**
  - `Y` & `37.0.12`

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by mikehardy** [comment]
Interesting
Unfortunately I can't imagine you will get any support for the "Fabric Crashlytics", you will need to migrate to current code ("Firebase Crashlytics" - just landed in react-native-firebase/crashlytics v8+) to make sure you have code that will receive attention

I wouldn't do anything until that was done as it's likely a time malinvestment

Then I would scan the upstream firebase-android-sdk issues repository to see if anyone else has seen anything similar. I'm not sure this module is doing much of anything with regard to the crashlytics APIs during init that should result in this behavior so it is likely an upstream issue.
You could dig into node_modules/@react-native-firebase/crashlytics after the upgrade and put some quick console logging around every underlying SDK API this module wraps/calls to see what's going on, and compare with the docs for the APIs to make sure the module isn't mis-using them somehow, as a possible avenue to move forward

**Comment by RodolfoGS** [comment]
Hi @mikehardy, thanks for your fast response, I'll close this issue until I update to v8+ and test if the issue persists, if I continue with the problems I'll create a new issue.

Thanks for your support, I really appreciate it.
